evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -0,0 +1,506 @@
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y)
+    X_t, y_t = imputer.transform(X, y)
+    X_expected = pd.DataFrame(np.array([[1, 2, 2, 0], [1, 0, 0, 0], [1, 0, 0, 0]]))
+    y_expected = pd.Series([0, 0, 1])
+    assert_frame_equal(X_t, X_expected, check_dtype=False)
+    assert_series_equal(y_t, y_expected, check_dtype=False)
+
+    imputer = TimeSeriesImputer()
+    X_t, y_t = imputer.fit_transform(X, y)
+    assert_frame_equal(X_t, X_expected, check_dtype=False)
+    assert_series_equal(y_t, y_expected, check_dtype=False)
+
+
+@pytest.mark.parametrize("data_type", ["np", "pd", "ww"])
+def test_imputer_empty_data(data_type, make_data_type):
+    X = pd.DataFrame()
+    y = pd.Series()
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+    X_expected = pd.DataFrame(index=pd.Index([]), columns=pd.Index([]))
+    y_expected = pd.Series()
+
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y)
+    X_t, y_t = imputer.transform(X, y)
+    assert_frame_equal(X_t, X_expected, check_dtype=False)
+    assert_series_equal(y_t, y_expected, check_dtype=False)
+
+    imputer = TimeSeriesImputer()
+    transformed, _ = imputer.fit_transform(X, y)
+    assert_frame_equal(transformed, X_expected, check_dtype=False)
+
+
+def test_imputer_does_not_reset_index():
+    X = pd.DataFrame(
+        {
+            "input_val": np.arange(10),
+            "target": np.arange(10),
+            "input_cat": ["a"] * 7 + ["b"] * 3,
+        }
+    )
+    X.loc[5, "input_val"] = np.nan
+    X.loc[5, "input_cat"] = np.nan
+    assert X.index.tolist() == list(range(10))
+    X.ww.init(logical_types={"input_cat": "categorical"})
+
+    X.drop(0, inplace=True)
+    y = X.ww.pop("target")
+
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y=y)
+    transformed, _ = imputer.transform(X)
+    pd.testing.assert_frame_equal(
+        transformed,
+        pd.DataFrame(
+            {
+                "input_val": [1.0, 2, 3, 4, 5, 6, 7, 8, 9],
+                "input_cat": pd.Categorical(["a"] * 6 + ["b"] * 3),
+            },
+            index=list(range(1, 10)),
+        ),
+    )
+
+
+def test_imputer_no_nans(imputer_test_data):
+    X = imputer_test_data[["categorical col", "object col", "bool col"]]
+    y = pd.Series([0, 0, 1, 0, 1] * 4)
+    imputer = TimeSeriesImputer(
+        categorical_impute_strategy="backwards_fill",
+        numeric_impute_strategy="forwards_fill",
+    )
+    imputer.fit(X, y)
+    X_t, y_t = imputer.transform(X, y)
+    X_expected = pd.DataFrame(
+        {
+            "categorical col": pd.Series(
+                ["zero", "one", "two", "zero", "two"] * 4, dtype="category"
+            ),
+            "object col": pd.Series(["b", "b", "a", "c", "d"] * 4, dtype="category"),
+            "bool col": [True, False, False, True, True] * 4,
+        }
+    )
+    assert_frame_equal(X_t, X_expected, check_dtype=False)
+    assert_series_equal(y_t, y, check_dtype=False)
+
+
+def test_imputer_with_none():
```

Comment:
It's probably not worth the effort, but any thoughts on modifying the `imputer_data` fixture we're using so that it can return different types of null values?